Posts/2021/02/Get-Services, PowerShell 7 VS Windows PowerShell 5.1

```diff
@@ -0,0 +1,150 @@
+RequiredServices          AliasProperty
+Disposed                          Event
+Close                            Method
+Continue                         Method
+CreateObjRef                     Method
+Dispose                          Method
+Equals                           Method
+ExecuteCommand                   Method
+GetHashCode                      Method
+GetLifetimeService               Method
+GetType                          Method
+InitializeLifetimeService        Method
+Pause                            Method
+Refresh                          Method
+Start                            Method
+Stop                             Method
+WaitForStatus                    Method
+CanPauseAndContinue            Property
+CanShutdown                    Property
+CanStop                        Property
+Container                      Property
+DependentServices              Property
+DisplayName                    Property
+MachineName                    Property
+ServiceHandle                  Property
+ServiceName                    Property
+ServicesDependedOn             Property
+ServiceType                    Property
+Site                           Property
+StartType                      Property
+Status                         Property
+ToString                   ScriptMethod
+```
+
+## PowerShell 7
+
+The big difference is in the Property, now in Powershell 7, it's possible to read some additional properties that were not available in Windows PowerShell 5.1, such as **UserName, BinaryPathName, StartType.** So let's see how to read the **Username**, **BinaryPathName,** and **StartType** using PowerShell 7
+
+```powershell
+PS 7> Get-Service workstation | select  Username,Starttype,BinaryPathName
+```
+
+The output is a clear, with all the required result with one-liner native command.
+
+```powershell
+UserName                    StartType BinaryPathName
+--------                    --------- --------------
+NT AUTHORITY\NetworkService Automatic C:\WINDOWS\System32\svchost.exe -k NetworkService -p
+```
+
+## Windows Powershell 5.1
+
+For Windows PowerShell 5.1, the operation is not straight forward, as such in PowerShell 7, we need to use the `Get-CimInstance` and pass the required WQL query, so in Windows PowerShell 5.1 run the following command
+
+```powershell
+WPS 5.1> Get-CimInstance -Query 'select * from Win32_Service where caption like "Workstation"' | select StartName,StartMode,PathName
+```
+
+```powershell
+StartName                   StartMode PathName
+---------                   --------- --------
+NT AUTHORITY\NetworkService Auto      C:\WINDOWS\System32\svchost.exe -k NetworkService -p
+```
+
+As shown in the command, we invoke `Get-CimInstance` (or if you are using Get-WmiObject) and wrote a query to get the service name, and after that select the required properties, which is a long way and require to know extra information related to the original service name and some basic WMI query.
```

Comment:
remove the mention to Get-WmiObject. 
Also not sure what the point of the final part of the sentence. is.

Reply:
Will remove it.
for the last part, I was trying to explain that with Windows PowerShell 5.1 an extra effort will be required. 
But I can remove it.